### ISSUE TITLE
Logging improvements

### ISSUE THREAD

**Issue by eriwen** [gradle/gradle#1815]
Log only "interesting" stuff by default, and give more options for users to see progressively more information.... and I'm not talking about different log levels here.

### Expected Behavior
```
[:util:compileJava]
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

[task execution listener]
Log message from build logic
```

### Current Behavior
```
:util:compileJava
Note: Some input files use unchecked or unsafe operations.
Log message from build logic
Note: Recompile with -Xlint:unchecked for details.
```

**Comment by awhawks** [comment]
I find --parallel output almost useless today in <= 3.5

Could have a prefix in front of each log line in a format like
[ <project name> <timestamp> ] <normall log output>
or even better
[ <project name> <timestamp> <task name> ] <normall log output>

So that we could identify which project did the logging.
This would greatly improve --parallel logging
This would allow post processing of logs to filter and find relevant debug data.

**Comment by awhawks** [comment]
I find --parallel output almost useless today in <= 3.5

Could have a prefix in front of each log line in a format like
[ 'ProjectA' 17:12:00.123  ] normal log output 
or even better
[ 'ProjectA' 17:12:00.123 'TaskA' ] normal log output 

So that we could identify which project did the logging.
This would greatly improve --parallel logging
This would allow post processing of logs to filter and find relevant debug data.

**Comment by lacasseio** [comment]
Thanks @awhawks for your comment on this. We are hard at work in improving the logging in the console specifically for parallel execution. I encourage you to keep an eye on Gradle 4.0 and 4.1 coming in the next couple months regarding this issue.

**Comment by eriwen** [comment]
@awhawks We had a lengthy discussion about a "prefix" vs. grouping headers, and decided that the prefix would be far too noisy in practice, given the length of our build operation descriptions (sometimes task names, sometimes not). We believe that grouping all logs for the same task together is preferable to a prefix, and allows sufficient scriptability (advanced use of awk for scraping instead of grep, but still very workable).

**Comment by eriwen** [comment]
This epic has no issues attached and was addressed in Gradle 3.5. Closing.
